src/processors/default/addMimeTypeToProcessedTracks.ts

```diff
@@ -0,0 +1,92 @@
+import _ from 'lodash';
+
+import { Table } from '../../db/db';
+import { MimeEnum } from '../../types/media';
+import { NFT } from '../../types/nft';
+import { Clients, Processor } from '../../types/processor';
+import { ProcessedTrack } from '../../types/track';
+import { Trigger } from '../../types/trigger';
+import { rollPromises } from '../../utils/rollingPromises';
+
+export enum SourceIPFS {
+  AUDIO = 'Audio',
+  ARTWORK = 'Artwork'
+}
+
+export type TrackNftJoin = ProcessedTrack & Partial<NFT>;
+
+const TIMEOUT = parseInt(process.env.METADATA_REQUEST_TIMEOUT!)
+const QUERY_LIMIT = process.env.QUERY_TRIGGER_BATCH_SIZE
+
+const missingMimeType: (source: SourceIPFS) => Trigger<undefined> = (source) => {
+  return async (clients) => {
+    // find approved tracks (and the nftID) which are missing a mimeType; skipping any which have had errors
+    const query = `select p.*, n.id as "nftId"
+      FROM "${Table.nfts}" as n
+      LEFT OUTER JOIN "${Table.nfts_processedTracks}" as j
+      ON n.id = j."nftId"
+      LEFT OUTER JOIN "${Table.processedTracks}" as p
+      ON j."processedTrackId" = p.id
+      LEFT OUTER JOIN "${Table.nftProcessErrors}" as e
+      ON n.id = e."nftId"
+      WHERE n.approved = true AND
+      e."processError" is NULL AND
+      e."metadataError" is NULL AND
+      p."lossy${source}MimeType" is NULL AND
+      p."lossy${source}IPFSHash" is NOT NULL
+      LIMIT ${QUERY_LIMIT}`
+
+    const processedTracks = (await clients.db.rawSQL(query)).rows;
+    return processedTracks;
+  }
+}
+
+export const addMimeTypeToProcessedTracks: (source: SourceIPFS) => Processor =
+  (source) => ({
+    name: `addLossy${source}MimeTypeToProcessedTracks`,
+    trigger: missingMimeType(source),
+    processorFunction: async (input: TrackNftJoin[], clients: Clients) => {
+      console.log(`Fetching processed track ${source} mime types`);
+
+      const updatedMimeTypes = async (trackNftJoin: any) => {
+        const ipfsHash = trackNftJoin[`lossy${source}IPFSHash`];
+        let response: any;
+        let errorMsg: string | undefined = undefined;
+        let contentType: any = '';
+
+        try {
+          response = await clients.axios.head(`${process.env.IPFS_ENDPOINT}${ipfsHash}`, { timeout: TIMEOUT })
+          contentType = response.headers['content-type']?.toLowerCase();
+        } catch (e: any) {
+          errorMsg = `Error: failed to fetch ${source} mime type for ipfs hash: ${ipfsHash} with error: ${e.message}`;
+        }
+
+        if (contentType && !Object.values(MimeEnum).includes(contentType)) {
+          errorMsg = `Error: invalid ${source} mime type: ${contentType} for ipfs hash: ${ipfsHash}`;
+        }
```

Comment:
i think we should explicitly check that:
 - AUDIO is audiovisual (ie, not an image)
 - ARTWORK is image/visual (ie, not pure audio)

Reply:
in the context of my other comment (https://github.com/spinamp/spindexer/pull/138/files#r1032509476), if we go with that solution, i think we should either:
 - have extra fields on the ipfsFiles table like isAudio and isVideo and isImage that we store and use to match against as part of the tracks view
 - match against the hardcoded array of mimetypes as part of the view to filter out mismatched audio/artwork

Reply:
@musnit just to make sure I understand correctly, the "matching" we're talking about is just updating the views for `processed_tracks` right?

At the moment, `processed_tracks` includes both `lossyArtworkIPFSHash` and `lossyAudioIPFSHash` where both are not null, so we'd update the `processed_tracks` view to ensure that we:

1. exclude `lossyArtworkIPFSHash` and `lossyAudioIPFSHash` that do not have a mime type
2. only return `processed_tracks` where `lossyArtworkIPFSHash` is image/video AND `lossyAudioIPFSHash` is audio/video

Is that in-line with what you're thinking?

Reply:
1 - yes, though exclude the whole track if either is missing a mime type, not just those fields
2 - yes

and yeah this sounds good. I suppose there may be some edge cases we uncover where we still actually want to return the track, but let's implement it like this and then see